measurecrack.py

```diff
@@ -0,0 +1,175 @@
+    
+    # resized = imutils.resize(mask, width=600)
+    # cv2.imshow("mask", resized)
+    # cv2.waitKey(0)
+    
+    # Detect lines
+    lines = cv2.HoughLines(mask, 1, np.pi/180, 700)
+    # If no lines are detected, revert to ratio method
+    if (lines is None):
+        return measure_crack(image)
+    a,b,c = lines.shape
+    
+    top = []
+    bottom = []
+    left = []
+    right = []
+    
+    height, width, channels = image.shape
+    
+    for i in range(a):
+        rho = lines[i][0][0]
+        theta = lines[i][0][1]
+        
+        # Sort lines into left, right, top, and bottom
+        if (theta < 0.2):
+            if (rho < 0.25 * width):
+                left.append([rho, theta])
+            elif (rho > 0.75 * width):
+                right.append([rho, theta])
+        elif (theta > 2.94):
+            if (rho > -0.25 * width):
+                left.append([rho, theta])
+            elif (rho < -0.75 * width):
+                right.append([rho, theta])
+        elif (theta > 1.37 and theta < 1.77):
+            if (rho < 0.25 * height):
+                top.append([rho, theta])
+            elif (rho > 0.75 * height):
+                bottom.append([rho, theta])
+    
+    # If grid line detection fails, revert to ratio method
+    if (len(top) == 0 or len(bottom) == 0 or len(left) == 0 or len(right) == 0):
+        return measure_crack(image)
+
+    x1, y1 = intersect(top, left)
+    x2, y2 = intersect(top, right)
+    x3, y3 = intersect(bottom, left)
+    x4, y4 = intersect(bottom, right)
+    
+    # Draw lines on image
+    # cv2.line(image, (x1, y1), (x2, y2), (0, 255, 0), 5)
+    # cv2.line(image, (x1, y1), (x3, y3), (0, 255, 0), 5)
+    # cv2.line(image, (x2, y2), (x4, y4), (0, 255, 0), 5)
+    # cv2.line(image, (x3, y3), (x4, y4), (0, 255, 0), 5)
+    
+    # Display image with lines drawn
+    # resized = imutils.resize(image, width=600)
+    # cv2.imshow("lines", resized)
+    # cv2.waitKey(0)
+    
+    # Find perspective transform matrix
+    src = np.float32([[x1, y1], [x2, y2], [x3, y3], [x4, y4]])
+    dst = np.float32([[0, 0], [3000, 0], [0, 3000], [3000, 3000]])
+    matrix = cv2.getPerspectiveTransform(src, dst)
+    
+    # Perspective transform original image
+    warped = cv2.warpPerspective(image, matrix, (3000, 3000))
+    
+    # Display warped image
+    # resized = imutils.resize(warped, width=600)
+    # cv2.imshow("warped", resized)
+    # cv2.waitKey(0)
+    
+    width, height = blue_rectangle(warped)
+    
+    # Find larger dimension. Divide by 100 because 100 pixels = 1 cm.
+    if (width > height):
+        ret = width / 100
+    else:
+        ret = height / 100
+    
+    # If dimension is out of bounds, revert to ratio method
+    if (ret > 21 or ret < 7):
+        return measure_crack(image)
+    else:
+        return ret
+
+# Find the average intersection point between two sets of lines    
+def intersect(lines1, lines2):
+    xs = []
+    ys = []
+    for line1 in lines1:
```

Comment:
you should comment what youre doing here more.is not clear. 

also, what are a1, b1, m1, etc.  please use better variable names